sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp

```diff
@@ -31,26 +31,51 @@ namespace Azure { namespace Core { namespace Http { namespace _internal {
    * @remark See #policy.hpp
    */
   class HttpPipeline final {
-  private:
+  protected:
+    std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> m_policies;
+
+  public:
+    /**
+     * @brief Construct HTTP pipeline with the sequence of HTTP policies provided.
+     *
+     * @param policies A sequence of #Azure::Core::Http::Policies::HttpPolicy
+     * representing a stack, first element corresponding to the top of the stack.
+     *
+     * @throw `std::invalid_argument` when policies is empty.
+     */
+    explicit HttpPipeline(
+        const std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>>& policies)
+    {
+      if (policies.size() == 0)
+      {
+        throw std::invalid_argument("policies cannot be empty");
+      }
+
+      m_policies.reserve(policies.size());
+      for (auto& policy : policies)
+      {
+        m_policies.emplace_back(policy->Clone());
+      }
+    }
+
     /**
      * @brief Construct a new HTTP Pipeline object from clientOptions.
      *
      * @remark The client options includes per retry and per call policies which are merged with the
      * service-specific per retry policies.
      *
      * @param clientOptions The SDK client options.
-     * @param telemetryServiceName The name of the service for sending telemetry.
-     * @param telemetryServiceVersion The version of the service for sending telemetry.
+     * @param telemetryPackageName The name of the service for sending telemetry.
+     * @param telemetryPackageVersion The version of the service for sending telemetry.
      * @param perRetryPolicies The service-specific per retry policies.
      * @param perCallPolicies The service-specific per call policies.
      */
     explicit HttpPipeline(
         Azure::Core::_internal::ClientOptions const& clientOptions,
```

Comment:
This name change was intended to correctly reflect the semantics of these two parameters based on the [User Agent Guidelines](https://azure.github.io/azure-sdk/general_azurecore.html#user-agent-format).

The field is *not* the service name, it's the package name. 

I am fixing it because I messed it up because of the incorrect parameter name (service name implies ARM service name, package name strictly implies package name).

Reply:
Renaming is fine, but is order the same? Can I compile older client library with the new core and vice versa? It's all that I was taking about.

Reply:
Oh, I see what happened - it's a confused diff.

There used to be 4 separate constructors for `HttpPipeline`. I removed two of them because they weren't used (the one without the package name and the common constructor used by the package name ctor and the non-package name ctor). But the order of the package name ctor parameters didn't change.

Reply:
That is great, and I like the change and support it, but we need to check that none of the previously released GA versions of any client library is using any of the 2 deleted syntaxes when compiling the SDK library.

Reply:
THe only clients which were affected where `AttestationClient` and `AttestationAdministrationCLient` and this PR fixes both of them.

Reply:
And to be clear: I have a set of outstanding questions to Liudmilla about this. The current pattern means that clients that support otel end up looking like:

```
m_tracingFactory = TracingFactory(options, "Microsoft.Attestation", “azure-security-attestation-cpp”,  PackageVersion::ToString());
m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(options, std::move(perRetryPolicies), std::move(perCallPolicies),
      “azure-security-attestation-cpp”, PackageVersion::ToString());
```

because the opentelemetry tracer name AND the user-agent both need the package name and package version. And it seems really silly (and very fragile) to have the exact same information specified twice.